testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java

```diff
@@ -2111,6 +2111,14 @@ void checkPeriodPredicate(Checker checker) {
             + "FROM `DEPT`))) AND (3 = 4))");
   }
 
+  /** Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-6986">[CALCITE-6986]
+   * Parser rejects SQL sources that produce an empty statement list</a>. */
+  @Test public void testEmpty() {
+    sql("").list().ok();
```

Comment:
I suggest adding new test cases to form a contrast. For example: 
```
sql("").ok()
sql(" ").ok()
sql("-- comment").ok()
```
Throwing an exception is also a valid test case.

Reply:
These are actually not very interesting test cases, because `sql(...).ok()` expects exactly ONE SQL statement, and there there isn't any in these strings. So these test cases are not testing this issue. @julianhyde made a comment about this in the Jira.